rllib/execution/parallel_requests.py

```diff
@@ -178,7 +180,237 @@ def wait_asynchronous_requests(
+                elif num_inflight_req_a1 > num_inflight_req_a2:
+                    actor = a2
+                else:
+                    if self._call_queue[a1].qsize() < self._call_queue[a2].qsize():
+                        actor = a1
+                    else:
+                        actor = a2
+            self._call_queue[actor].put(call)
+
+        self._run()
+
+    def _run(self) -> int:
+        """Launch the submited requests remotely on workers that don't have more
+        than max_inflight_requests in flight.
+        """
+        num_requests_launched = 0
+        for actor, q in self._call_queue.items():
+            # print(len(self._remote_requests_in_flight[actor]))
+            # print(self._max_remote_requests_in_flight)
+            while (
+                len(self._remote_requests_in_flight[actor])
+                < self._max_remote_requests_in_flight
+            ) and q.qsize() > 0:
+                remote_fn, fn_args, fn_kwargs = q.get()
+                req = actor.apply.remote(remote_fn, *fn_args, **fn_kwargs)
+                self._remote_requests_in_flight[actor].add(req)
+                self._pending_to_actor[req] = actor
+                self._pending_remotes.append(req)
+                num_requests_launched += 1
+        return num_requests_launched
+
+    def get_ready_results(self) -> Dict[ActorHandle, List[Any]]:
+        """Get results that are ready to be returned
+
+        Returns:
+            A dictionary of actor handles to lists of returns from tasks that were
+             previously submitted to this actor pool that are now ready to be returned.
+             If return_object_refs
+
+        """
+        ready_requests_dict = defaultdict(list)
+        ready_requests, self._pending_remotes = ray.wait(
+            self._pending_remotes,
+            timeout=0,
+            num_returns=len(self._pending_remotes),
+        )
+        if not self._return_object_refs:
+            objs = ray.get(ready_requests)
+        else:
+            objs = ready_requests
+        for req, obj in zip(ready_requests, objs):
+            actor = self._pending_to_actor[req]
+            self._remote_requests_in_flight[actor].remove(req)
+            ready_requests_dict[actor].append(obj)
+            del self._pending_to_actor[req]
+        del ready_requests
+        self._run()
+        return dict(ready_requests_dict)
+
+    def add_worker(self, new_worker: ActorHandle) -> None:
+        """Add a new worker to the manager
+
+        Args:
+            new_worker: The actor to add
+
+        """
+        if new_worker in self._all_workers:
+            return
+        self._all_workers.add(new_worker)
+
+    def remove_worker(self, worker: ActorHandle) -> None:
+        """Remove a worker from the manager
+
+        Args:
+            worker: The actor to remove
+        """
+        if worker not in self._all_workers:
+            return
+        self._all_workers.remove(worker)
```

Comment:
you can actually just put a. worker actor in lame duck mode. (not receiving new calls, only waiting for existing calls to finish)
then actual deletion can happen in _run() function.

Reply:
yeah this is a good idea actually I will do this instead.